stdlib/public/runtime/SwiftRT-ELF.cpp

```diff
@@ -34,41 +34,53 @@ DECLARE_SWIFT_SECTION(swift5_fieldmd)
 DECLARE_SWIFT_SECTION(swift5_assocty)
 DECLARE_SWIFT_SECTION(swift5_replace)
 DECLARE_SWIFT_SECTION(swift5_replac2)
+DECLARE_SWIFT_SECTION(swift5_builtin)
+DECLARE_SWIFT_SECTION(swift5_capture)
 }
 
 #undef DECLARE_SWIFT_SECTION
 
-namespace {
-static swift::MetadataSections sections{};
-}
-
-__attribute__((__constructor__))
-static void swift_image_constructor() {
 #define SWIFT_SECTION_RANGE(name)                                              \
   { reinterpret_cast<uintptr_t>(&__start_##name),                              \
     static_cast<uintptr_t>(&__stop_##name - &__start_##name) }
 
-  sections = {
-      swift::CurrentSectionMetadataVersion,
-      0,
+static swift::MetadataSections sections{
+    swift::CurrentSectionMetadataVersion,
+    0,
 
-      nullptr,
-      nullptr,
+    nullptr,
+    nullptr,
 
-      SWIFT_SECTION_RANGE(swift5_protocols),
-      SWIFT_SECTION_RANGE(swift5_protocol_conformances),
-      SWIFT_SECTION_RANGE(swift5_type_metadata),
+    SWIFT_SECTION_RANGE(swift5_protocols),
+    SWIFT_SECTION_RANGE(swift5_protocol_conformances),
+    SWIFT_SECTION_RANGE(swift5_type_metadata),
 
-      SWIFT_SECTION_RANGE(swift5_typeref),
-      SWIFT_SECTION_RANGE(swift5_reflstr),
-      SWIFT_SECTION_RANGE(swift5_fieldmd),
-      SWIFT_SECTION_RANGE(swift5_assocty),
-      SWIFT_SECTION_RANGE(swift5_replace),
-      SWIFT_SECTION_RANGE(swift5_replac2),
-  };
+    SWIFT_SECTION_RANGE(swift5_typeref),
+    SWIFT_SECTION_RANGE(swift5_reflstr),
+    SWIFT_SECTION_RANGE(swift5_fieldmd),
+    SWIFT_SECTION_RANGE(swift5_assocty),
+    SWIFT_SECTION_RANGE(swift5_replace),
+    SWIFT_SECTION_RANGE(swift5_replac2),
+    SWIFT_SECTION_RANGE(swift5_builtin),
+    SWIFT_SECTION_RANGE(swift5_capture),
+};
```

Comment:
Does this not create a constructor?  There is now a C++ constructor which runs separately than the actual constructor?

Reply:
Yeah. Maybe it gets optimized away by LLVM GlobalOpt, but we should double-check. What I had meant in the previous PR was that we could generate the symbol and its definition in a global `asm` block, since the section ranges can be represented as constant expressions in assembly language, but not C.